File: src/windows/AmbientContextMcp.Core/Models/EventSchema.cs
namespace AmbientContextMcp.Core.Models;

/// <summary>
/// イベント単位の payload スキーマを記述する。<c>ambient_context_describe_events</c> ツール経由で
/// クライアントが「foreground_changed の payload には何が入るか」「media_session_changed の title は高機微か」
/// などを 1 回で参照できるようにする。
/// </summary>
public sealed class EventSchema
{
    public string Name { get; init; } = "";

    public string Sensitivity { get; init; } = "low";

    public string Description { get; init; } = "";

    public IReadOnlyList<EventPayloadKey> PayloadKeys { get; init; } = [];
}

public sealed class EventPayloadKey
{
    public string Key { get; init; } = "";

    public string Sensitivity { get; init; } = "low";

    public string Description { get; init; } = "";

    public string Example { get; init; } = "";
}

File: src/macos/Sources/AmbientContextCore/Models/EventSchema.swift
import Foundation

/// イベント単位の payload スキーマを記述する。`ambient_context_describe_events` ツール経由で
/// クライアントが「フォアグラウンドアプリ切替 (foreground_changed) の payload には何が入るか」
/// 「media_session_changed の title は高機微か」などを 1 回で参照できるようにする。
public struct EventSchema: Codable, Sendable, Hashable {
    public var name: String
    public var sensitivity: String
    public var description: String
    public var payloadKeys: [EventPayloadKey]

    public init(
        name: String = "",
        sensitivity: String = "low",
        description: String = "",
        payloadKeys: [EventPayloadKey] = []
    ) {
        self.name = name
        self.sensitivity = sensitivity
        self.description = description
        self.payloadKeys = payloadKeys
    }
}

public struct EventPayloadKey: Codable, Sendable, Hashable {
    public var key: String
    public var sensitivity: String
    public var description: String
    public var example: String

    public init(
        key: String = "",
        sensitivity: String = "low",
        description: String = "",
        example: String = ""
    ) {
        self.key = key
        self.sensitivity = sensitivity
        self.description = description
        self.example = example
    }
}
